add upgrade free space check to document (#332)
harvester/docs: docs/upgrade/automatic.md

    - Back up your VMs if needed.
- Do not operate the cluster during an upgrade. For example, creating new VMs, uploading new images, etc.
- Make sure your hardware meets the **preferred** [hardware requirements](../index.md#hardware-requirements). This is due to there will be intermediate resources consumed by an upgrade.

> - Make sure each node has at least 30 GiB of free space (`df -h /usr/local/`). If any node in the cluster has fewer free space, the upgrade will be denied. Check [free space requirment](#free-space-requirement) for more information.

Reviewer: ```suggestion
- Make sure each node has at least 30 GiB of free space (`df -h /usr/local/`). If any node in the cluster has fewer free space, the upgrade will be denied. Check [free space requirement](#free-space-requirement) for more information.
```

> - Make sure each node has at least 30 GiB of free space (`df -h /usr/local/`). If any node in the cluster has fewer free space, the upgrade will be denied. Check [free space requirement](#free-space-requirement) for more information.

Reviewer 2: ```suggestion
- Make sure each node has at least 30 GiB of free space (`df -h /usr/local/`). If any node in the cluster has less than 30 GiB of free space, the upgrade will be unsuccessful. Check [free space requirement](#free-space-requirement) for more information.
```

Author: @LucasSaintarbor Here a window is poped immediately to say upgrade is denied. It is more accurate than `unsuccessful`.

    ```

- An upgrade button should show up on the Harvester GUI Dashboard page.

## Free space requirement

_Available as of v1.2.0_

> The minimal free space requirement in Harvester v1.2.0 is 30Gib, and it will be revised in each release.

Reviewer 2: ```suggestion
The minimum free space requirement in Harvester v1.2.0 is 30 Gib, which will be revised in each release.
```

> Harvester will check the free space of each node when clicking `Upgrade` button, if any node does not meet the requirement, the upgrade will be denied like below:

Reviewer 2: ```suggestion
Harvester will check the amount of free space on each node when you select **Upgrade**. If any node does not meet the requirement, the upgrade will be denied as follows:
```

![](/img/v1.2/upgrade/upgrade_free_space_check.png)

In certain cases, some nodes may have less free disk space, but you still want to try the upgrade, you may customize it via updating the `harvesterhci.io/minFreeDiskSpaceGB` annotation of `Version` object.

```
apiVersion: harvesterhci.io/v1beta1
kind: Version
metadata:
  annotations:
    harvesterhci.io/minFreeDiskSpaceGB: "30" # the value is pre-defined and may be customized
  name: 1.2.0
  namespace: harvester-system
spec:
  isoChecksum: <SHA-512 checksum of the ISO>
  isoURL: http://192.168.0.181:8000/harvester-master-amd64.iso
  minUpgradableVersion: 1.1.2
  releaseDate: "20230609"
```

:::caution

> - Setting a smaller value than the pre-defined one may cause upgrade fail in the middle state. It is not recommended in the production environment.

Reviewer 2: ```suggestion
Setting a smaller value than the pre-defined value may cause the upgrade to fail and is not recommended in a production environment.
```

> 

Author: updated, thanks.
